### DISCUSSION TITLE
Blob and File seem to be missing?

### DISCUSSION THREAD

**Discussion by wmertens** [nodejs/help#4923] [Q&A] [answered]
When using FormData to upload a file in the browser, you can pass it a File which is actually a Blob and that can lazily represent a huge file.

This isn't possible in Node, you need to use something like https://github.com/form-data/form-data to work with memory efficiently.

Is Blob support planned?

**Comment by Trott** [comment, marked as answer]
Blob was added in Node.js 18.

**Comment by wmertens** [comment]
Argh, totally missed that, sorry. "blob" and "file" are hard to search on :sweat_smile:.

But `File` is still missing, right? Or is there some other way to lazily access a file  as a Blob?

**Reply by aduh95** [reply]
The [`File`](https://developer.mozilla.org/en-US/docs/Web/API/File) class has not been implemented in Node.js (and there doesn't seem to be a good use case for it in Node.js). As of Node.js v16.7.0, you can create a `Blob` that contains the file content using the following utility:

```js
import { createReadStream } from 'node:fs';
import { blob } from 'node:stream/consumers';

const myFile = await blob(createReadStream('path/to/my/file'));
```

**Reply by wmertens** [reply]
Hmm but that will load the entire file into memory https://github.com/nodejs/node/blob/27ecf1d94cc0c6eaae09e30b3c8a1251d1b4dd46/lib/stream/consumers.js#L29-L34

The Blob API should make it possible to append a 2GB file to a FormData instance without loading it into memory, right? Is that something that Node can support?

**Reply by targos** [reply]
I'm in favor of having an implementation of `File` in Node.js. We just need a volunteer to do it.
The issue for it is https://github.com/nodejs/node/issues/39015
